### PR TITLE
Fix TPU Splash local sliding window size

### PR DESCRIPTION
**Pull request by huytransformer**
# Description

Dense local sliding keeps exactly `sliding_window_size` tokens. Splash `LocalMask` uses an inclusive left window, so passing `sliding_window_size` keeps 1 extra token. This changes the Splash window to `sliding_window_size - 1`.

# Tests
Ran successfully
```bash
python3 -m pytest tests/unit/attention_test.py -k SplashLocalMaskTest
```

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [x] I have performed a self-review of my code. For an optional AI review, add the `gemini-review` label.
- [x] I have necessary comments in my code, particularly in hard-to-understand areas.
- [x] I have run end-to-end tests tests and provided workload links above if applicable.
- [x] I have made or will make corresponding changes to the doc if needed, including adding new documentation pages to the relevant Table of Contents (toctree directive) as explained in [our documentation](https://maxtext.readthedocs.io/en/latest/development.html#adding-new-documentation-files).

### CODE REVIEW

File: src/maxtext/layers/attention_op.py [hunk 1]
```diff
@@ -1286,7 +1286,7 @@ def create_sa_config(config, query, key, attn_logits_soft_cap):
         raise ValueError("Sliding_window_size must be set if Local Sliding attention type")
       mask &= mask_module.LocalMask(
           shape=(query.shape[2], key.shape[2]),
-          window_size=(self.sliding_window_size, self.sliding_window_size),
+          window_size=(self.sliding_window_size - 1, 0),
```
Comment (NuojCheng) [inline src/maxtext/layers/attention_op.py]: I am not fully confident this change is valid. Why it is 0 for the 2nd index? cc @gagika for wisdom on local mask

Comment (huytransformer) [reply src/maxtext/layers/attention_op.py]: because when we logical AND with causal mask the right side doesn't matter; we could leave the right side as it was if that would make things more clear

Comment (huytransformer) [reply src/maxtext/layers/attention_op.py]: Done : )

**Review by igorts-git — APPROVED** [review]

File: src/maxtext/layers/attention_op.py (hunk 1 above)
Comment (aireenmei) [reply src/maxtext/layers/attention_op.py]: The Gemma4 implementation in Tunix (https://github.com/google/tunix/blob/main/tunix/models/gemma4/model.py#L925-L929) uses window_size=(self.sliding_window_size - 1, 0)

**Review by NuojCheng — APPROVED** [review]
